code/iaas/api-logic/src/test/java/io/cattle/platform/iaas/api/host/HostTemplateValidationFilterTest.java

```diff
@@ -0,0 +1,165 @@
+package io.cattle.platform.iaas.api.host;
+
+import io.cattle.platform.core.constants.HostTemplateConstants;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManager;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.cattle.platform.iaas.api.host.HostTemplateValidationFilter.*;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+
+public class HostTemplateValidationFilterTest {
```

Comment:
Generally, we write tests like this at the integration level, in python in the tess/inegration directory